scipy/stats/stats.py

```diff
@@ -214,6 +214,9 @@ def _broadcast_shapes_with_dropped_axis(a, b, axis):
     return shp
 
 
+@_axis_nan_policy_factory(
+        lambda x: x, n_samples=1, n_outputs=1, too_small=0,
+        result_unpacker=lambda x: (x,))
```

Comment:
Originally the decorator was designed for use with hypothesis tests, which typically output a namedtuple or other result object with two attributes: `statistic` and `pvalue`. It could be applied to statistics that returned scalar (or single n-d array) output by writing the `result_object` function to accept two arguents and simply return the first argument. For example, applying the decorator to `gmean` would look like:
```
@axis_nan_policy_factory(
        lambda x, y=None: x,  # this is what I mean 
        n_samples=1, too_small=0,
        result_unpacker=lambda x: (x,))
```

That is one way to do it. Here, though, I've added an argument to the wrapper factory, 
`n_output`, so that the correct number of outputs are passed into the `result_object`. So now, the result object here is simply:
```
lambda x: x
```
instead of 
```
lambda x, y=None: x
```
do we like that? I think it's a little more logical, but it requires passing an extra argument into `_axis_nan_policy_factory` rather than just writing `result_object` accordingly.

Reply:
I think using ``n_output`` is the more sensible solution. I don't think requiring an extra argument is too much of an issue as these wrappers are only used once per function and, once the initial work is done, won't need to be changed. This also seems more generic in case of any statistics using 3+ ouputs (I don't believe there are any but I could be wrong).

Reply:
There are some, yeah. I don't think that a scalar NaN is the appropriate output for some of the attributes (e.g. could be tuple) in most of those cases, but then `result_object` can handle it.